docs/GraphRepresentation.md

```diff
@@ -0,0 +1,321 @@
+the proportion of aggregate relevance derived from add-on A.
+
+
+This treatment is implemented as [`RowSum`](../taar_lite/recommenders/treatments.py#L64).
+
+
+### Proportional total normalization
+
+This treatment is a rescaled version of the [total relevance normalization](#total-relevance-normalization).
+Rather than total installs, popularity is quantified in terms of
+the proportion of coinstalls associated with each add-on.
+Here, the assumption is that more relevant recommendations tend to account for
+a higher proportion out of the given add-on's coinstallations than it does
+for other add-ons.
+
+Given add-on A, the coinstallation count for (A,B) for each add-on B
+is first divided by the sum of all A's coinstallation counts.
+The total relevance normalization is then applied to the resulting proportions.
+
+Given the adjacency matrix $C$,
+the treatment first divides each each entry $C_{ab}$
+by the row sum for row $a$,
+and then divides each entry by the column sum for column $b$
+in the resulting matrix.
+In terms of the graph representation,
+the weight on each edge (A,B) leaving A is first divided by
+the sum of the weights on all edges leaving A,
+and then the resulting edge weight is divided by
+the sum of the weights on all edges entering B.
+
+Intuitively, the first step converts coinstallation counts for A
+to the proportion of coinstallations of A allocated to each add-on B.
+In the case of general relevance scores, we can think of the row sum for A
+as a __relevance budget__,
+ie. the total amount of relevance it has available to assign to other add-ons.
+The first division has the effect of normalizing each add-on's
+relevance budget to 1.
+The next step then normalizes each coinstalled add-on's
+resulting aggregate relevance to 1.
+
+This treatment is implemented as [`RowNormSum`](../taar_lite/recommenders/treatments.py#L126).
+
+
+## Graph pruning
+
+Another class of treatments we may wish to apply involves modifying
+the vertices or edges of the relational graph themselves,
+possibly as a function of the edge weights.
+For example, we may wish to remove vertices or edges that fall below
+a minimum relevance threshold, in order to avoid recommending rare add-ons.
+Also, we may wish to ensure certain properties of the graph as a whole.
+It is possible for a user visiting AMO to "walk" the recommendation graph
+by clicking through a chain of subsequent recommendations.
+Although the recommendations themselves are memoryless,
+in that they only depend on the current add-on,
+we are interesting in optimizing the quality of experience
+for such users as well,
+such as not returning to a previous add-on in a short number of steps.
+
+
+### Minimum relevance pruning
+
+This treatment removes add-on vertices whose relevance score is lower
+than a given threshold,
+under the assumption that candidates with such low relevance
+would not make good recommendations, even if there no candidates
+with higher relevance.
```

Comment:
My understanding for the rationale was that it's privacy preserving.

Reply:
I added a comment about this. I'm less excited about highlighting the privacy aspect since it's not really a quantifiable guarantee.